### PR TITLE
Hour v2

### PR DESCRIPTION
**Pull request by laander**
@vistik

### CODE REVIEW

**Comment by vistik** [comment]
:8ball:

File: test/configLoading.spec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+jasmine.getFixtures().fixturesPath = 'base/test/fixtures';
+
+var baseConfig = require('./utils/defaultConfig');
```
Comment (laander) [inline test/configLoading.spec.js]: Codacy Issue found: ["baseConfig" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2386254823/issues/source?bid=2969108&fileBranchId=3228106#l5)

File: src/main.js [hunk 2]
```diff
@@ -571,17 +598,49 @@ function TimekitBooking() {
...
+  };
+
+  // Load config from remote (embed or hosted)
+  var loadRemoteConfig = function(suppliedConfig) {
+
+    config = setConfigDefaults(suppliedConfig)
+    timekitSetupConfig();
+    if (suppliedConfig.widgetId) {
+      return timekit
+      .getEmbedWidget({ id: suppliedConfig.widgetId })
```
Comment (laander) [inline src/main.js]: Codacy Issue found: ["response" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2386254816/issues/source?bid=2969108&fileBranchId=3228106#l623)

File: src/main.js [hunk 3]
```diff
@@ -470,30 +481,42 @@ function TimekitBooking() {
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
+    if (config.widgetSlug) campaignName = 'hosted-widget'
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2386254816/issues/source?bid=2969108&fileBranchId=3228106#l487)

File: src/main.js [hunk 4]
```diff
@@ -470,30 +481,42 @@ function TimekitBooking() {
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2386254816/issues/source?bid=2969108&fileBranchId=3228106#l486)

File: test/configLoading.spec.js (hunk 1 above)
Comment (laander) [inline test/configLoading.spec.js]: Codacy Issue found: ["baseConfig" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2386876815/issues/source?bid=2969108&fileBranchId=3228106#l5)

File: src/main.js (hunk 2 above)
Comment (laander) [inline src/main.js]: Codacy Issue found: ["response" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2386876808/issues/source?bid=2969108&fileBranchId=3228106#l623)

File: src/main.js (hunk 3 above)
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2386876808/issues/source?bid=2969108&fileBranchId=3228106#l487)

File: src/main.js (hunk 4 above)
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2386876808/issues/source?bid=2969108&fileBranchId=3228106#l486)

File: test/configLoading.spec.js (hunk 1 above)
Comment (laander) [inline test/configLoading.spec.js]: Codacy Issue found: ["baseConfig" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2419776782/issues/source?bid=2969108&fileBranchId=3228106#l5)

File: src/main.js [hunk 5]
```diff
@@ -449,51 +452,58 @@ function TimekitBooking() {
...
+    .include('attributes')
     .createBooking(args);
 
   };
 
   // Render the powered by Timekit message
   var renderPoweredByMessage = function(pageTarget) {
 
+    var campaignName = 'widget'
+    var campaignSource = window.location.hostname.replace(/\./g, '-')
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2419776775/issues/source?bid=2969108&fileBranchId=3228106#l473)

File: src/main.js [hunk 6]
```diff
@@ -449,51 +452,58 @@ function TimekitBooking() {
...
+    if (config.widgetSlug) campaignName = 'hosted-widget'
```
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2419776775/issues/source?bid=2969108&fileBranchId=3228106#l474)

File: src/main.js [hunk 7]
```diff
@@ -571,17 +585,49 @@ function TimekitBooking() {
...
+  };
+
+  // Load config from remote (embed or hosted)
+  var loadRemoteConfig = function(suppliedConfig) {
+
+    config = setConfigDefaults(suppliedConfig)
+    timekitSetupConfig();
+    if (suppliedConfig.widgetId) {
+      return timekit
+      .getEmbedWidget({ id: suppliedConfig.widgetId })
```
Comment (laander) [inline src/main.js]: Codacy Issue found: ["response" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2419776775/issues/source?bid=2969108&fileBranchId=3228106#l610)

File: src/main.js (hunk 5 above)
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2419896676/issues/source?bid=2969108&fileBranchId=3228106#l473)

File: src/main.js (hunk 6 above)
Comment (laander) [inline src/main.js]: Codacy Issue found: [Expected '{' and instead saw 'campaignName'.](https://www.codacy.com/app/laander/booking-js/file/2419896676/issues/source?bid=2969108&fileBranchId=3228106#l474)

File: src/main.js (hunk 7 above)
Comment (laander) [inline src/main.js]: Codacy Issue found: ["response" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2419896676/issues/source?bid=2969108&fileBranchId=3228106#l610)

File: test/configLoading.spec.js (hunk 1 above)
Comment (laander) [inline test/configLoading.spec.js]: Codacy Issue found: ["baseConfig" is defined but never used](https://www.codacy.com/app/laander/booking-js/file/2419896683/issues/source?bid=2969108&fileBranchId=3228106#l5)

**Pull request merged**